### PR TITLE
Migrate all SelectableText to SelectionArea.

### PR DESCRIPTION
**Pull request by olegzaim**
Replaced all SelectableText to SelectionArea.
Fixes https://github.com/flutter/gallery/issues/698

### CODE REVIEW

**Review by guidezpl — CHANGES_REQUESTED** [review]
This is not quite what is required. Only ~5 `SelectionArea`s should be necessary, one per `MaterialApp`, and one for the setting page, each placed near the top of the widget tree.

Each `SelectableText` should be replaced by `Text`.

File: lib/studies/shrine/app.dart [hunk 1]
```diff
@@ -164,7 +164,7 @@ class _ShrineAppState extends State<ShrineApp>
           initialRoute: ShrineApp.loginRoute,
           routes: {
             ShrineApp.loginRoute: (context) => const LoginPage(),
```
Comment (guidezpl) [inline lib/studies/shrine/app.dart:166]: should work on the login page as well

File: lib/studies/shrine/app.dart [hunk 2]
```diff
@@ -164,7 +164,7 @@ class _ShrineAppState extends State<ShrineApp>
...
-            ShrineApp.homeRoute: (context) => home,
+            ShrineApp.homeRoute: (context) => SelectionArea(child: home),
```
Comment (guidezpl) [inline lib/studies/shrine/app.dart:167]: is there an `ApplyTextOptions` missing here? if not, can it be moved here for consistency?

**Comment by olegzaim** [comment]
> * Rally text selection is invisible.

Hello, there is something wrong with the cursor for selection area because when I select all area it isn't invisible, but when I just tap and press it is stay invisible: 
- Single select with long press
long press for select
- Select all
select all

> * Demos don't have text selection

Is it enough or it's need to be selectable also inside demos?
home and demos select

**Comment by guidezpl** [comment]
I made some small fixes for consistency. The only thing missing I think is adding selection for the demos (material, cupertino, etc.)

**Comment by olegzaim** [comment]
> I made some small fixes for consistency. The only thing missing I think is adding selection for the demos (material, cupertino, etc.)

Okay, great thank you :)

**Review by guidezpl — APPROVED** [review]
Thanks for contributing!

**Comment by guidezpl** [comment]
@chunhtai This change triggers errors that cause  benchmark and golden tests to fail, would appreciate you taking a look if possible.

**Comment by guidezpl** [comment]
@olegzaim Could you take a look at the merge conflicts so that we can land this? Thanks!

**Comment by guidezpl** [comment]
Realized you might only have access to @OlegZaimNLT, if so, please create a new PR cloning this branch

**Comment by OlegZaimNLT** [comment]
Hey @guidezpl I can't merge because it says that I don't have permissions
Screenshot 2022-06-30 at 11 59 16
.

**Pull request closed without merging**